File: JULY-2025/31-07-2025/p898.java
class Solution {
    public int subarrayBitwiseORs(int[] arr) {
        Set<Integer> ans = new HashSet<>();
        Set<Integer> prev = new HashSet<>();
        for(int x : arr) {
            Set<Integer> cur = new HashSet<>();
            for(int y : prev) {
                cur.add(y | x);
            }
            cur.add(x);
            prev = cur;
            ans.addAll(cur);
        }

        return ans.size();
    }
}

File: JULY-2025/31-07-2025/p898.rs
use std::collections::HashSet;
use std::iter::once;

impl Solution {
    pub fn subarray_bitwise_o_rs(arr: Vec<i32>) -> i32 {

        let mut res  = HashSet::<i32>::new();
        let mut cur = HashSet::new();

        for &x in &arr {
            cur = cur.iter().map(|&y| y | x).chain(once(x)).collect();
            res.extend(&cur);
        }

        res.len() as _
    }
}